pulp_container/tests/functional/api/test_sync.py

```diff
@@ -1,231 +1,126 @@
-
-        # Sync the repository.
-        sync_response = repository_api.sync(repository.pulp_href, repository_sync_data)
-        monitor_task(sync_response.task)
-        repository = repository_api.read(repository.pulp_href)
-        self.assertIsNotNone(repository.latest_version_href)
-
-        # Sync the repository again.
-        latest_version_href = repository.latest_version_href
-        sync_response = repository_api.sync(repository.pulp_href, repository_sync_data)
-        monitor_task(sync_response.task)
-        repository = repository_api.read(repository.pulp_href)
-        self.assertEqual(latest_version_href, repository.latest_version_href)
-
-
-class SyncInvalidURLTestCase(PulpTestCase):
-    """Sync a repository with an invalid url on the Remote."""
-
-    def test_all(self):
-        """
-        Sync a repository using a Remote url that does not exist.
+import pytest
+from pulpcore.tests.functional import PulpTaskError
 
-        Test that we get a task failure.
-
-        """
-        client_api = gen_container_client()
-
-        repository_api = RepositoriesContainerApi(client_api)
-        repository = repository_api.create(ContainerContainerRepository(**gen_repo()))
-        self.addCleanup(repository_api.delete, repository.pulp_href)
-
-        remote_api = RemotesContainerApi(client_api)
-        remote_data = gen_container_remote(url="http://i-am-an-invalid-url.com/invalid/")
-        remote = remote_api.create(ContainerContainerRemote(**remote_data))
-        self.addCleanup(remote_api.delete, remote.pulp_href)
-
-        repository_sync_data = ContainerRepositorySyncURL(remote=remote.pulp_href)
-        sync_response = repository_api.sync(repository.pulp_href, repository_sync_data)
-
-        with self.assertRaises(PulpTaskError) as context:
-            monitor_task(sync_response.task)
-        self.assertIsNotNone(context.exception.task.error["description"])
+from pulp_container.tests.functional.utils import gen_container_remote
+from pulp_container.tests.functional.constants import PULP_FIXTURE_1
 
+from pulpcore.client.pulp_container import ContainerContainerRemote
 
-class TestRepeatedSync(PulpTestCase):
-    """Test behavior when a sync is repeated."""
 
-    @classmethod
-    def setUpClass(cls):
-        """Create class-wide variables."""
-        cls.client_api = gen_container_client()
+@pytest.fixture
+def synced_repository_with_filtered_tags(
+    container_repository_factory, container_remote_api, gen_object_with_cleanup, sync
+):
+    def _synced_repository_with_filtered_tags(include_tags=None, exclude_tags=None):
+        """Sync a new repository with the included tags passed as an argument."""
+        remote_data = gen_container_remote(
+            upstream_name=PULP_FIXTURE_1,
+            include_tags=include_tags,
+            exclude_tags=exclude_tags,
+        )
+        remote = gen_object_with_cleanup(container_remote_api, remote_data)
```

Comment:
Can this be `container_remote_factory(upstream_name=PULP_FIXTURE_1, include_tags=include_tags, exclude_tags=exclude_tags)` ?